pkg/kotsadmconfig/validation/regex.go

```diff
@@ -0,0 +1,40 @@
+package validation
+
+import (
+	"regexp"
+
+	"github.com/pkg/errors"
+	kotsv1beta1 "github.com/replicatedhq/kots/kotskinds/apis/kots/v1beta1"
+	configtypes "github.com/replicatedhq/kots/pkg/kotsadmconfig/types"
+)
+
+const (
+	regexMatchError = "Value does not match regex"
+)
+
+type regexValidator struct {
+	*kotsv1beta1.RegexValidator
+}
+
+func (v *regexValidator) Validate(input string) (*configtypes.ValidationError, error) {
+	regex, err := regexp.Compile(v.Pattern)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Invalid regex pattern. failed to compile regex")
+	}
+
+	matched := regex.MatchString(input)
+	if !matched {
+		return buildRegexValidationError(v.Message, regexMatchError), nil
+	}
+	return nil, nil
+}
+
+func buildRegexValidationError(validatorMessage string, errorMsg string) *configtypes.ValidationError {
```

Comment:
this function isn't needed, just return the message defined by the vendor

Reply:
If the vendor's message is empty, we need a default response

Reply:
https://github.com/replicatedhq/kots/blob/281e12f35ca7b706afd9b25aa067840ef7f874dc/pkg/kotsadmconfig/validation/regex.go#L26-L33

added inline code to use default message